### ISSUE TITLE
feature request for visualizing multiday recordings in a trial based manner

### ISSUE THREAD

**Issue by zcq0730** [MouseLand/suite2p#336]
Thanks a lot for sharing the wonderful software! The processing speed is blaze fast, and the GUI interface is really elegant and easy to use!
I like the visualization GUI in suite2P very much. My request is regarded to add some features to that part for visualizing multiday recordings in a trial based manner.

I run multiday behavior recordings from the same field of view. Each day of recording we call it a session, and each session contains multiple recording trials in which animals receive an external stimulus, eg. whisker deflection or auditory sound.

For extracting ROI activities from these multiday recordings, I followed the recommendation: https://mouseland.github.io/suite2p/_build/html/multiday.html. Just in order to keep the ROI assignment constant. So that we could compare the changes over experiment sessions or trials.

Before formal analysis, it is always good to have some basic understanding of the data. So visualization is important. However, because I concatenated the recordings over multiday, I could not **easily** find the frames that correspond to certain trials. I tried to write a GUI on my own to visualize the extracted ROI data in a trial based manner. It took me a long time and it works not very well. The visualization GUI within suit2P is perfect for this purpose. If some additional features could be implemented, that would be really cool!

Feature 1:
(control ZOOM IN area) Easy navigation trial corresponded frame range by additional dropbox or something like that, which basically only translate a particular trial into a range of frame number.  For example, Session 2, Trial 50, corresponds to frame range [1800 - 2020]. (The translation information is expected to be imported via the metadata import button.)
![feature_request](https://user-images.githubusercontent.com/40316783/82558335-b6de8b80-9b6d-11ea-915c-caa908688b2b.PNG)

Feature 2:
The calculation of heatmap and PC is now based on the whole length of concatenated recordings. Can it be calculated only based on frames from a session or even a trial? (Calculation based on ZOOM IN area) 

Feature 3:
When on press show selected cells in GUI, the corresponding selected ROIs within the ZOOM IN area will light up in the main GUI window. However, by default the X-axis does not correspond to the ZOOM IN area range. Could it be synchronized?

Thanks in advance, it would be really great to have these features. I think these features could be potentially useful to many others as well.

**Comment by carsen-stringer** [comment]
I get what you are saying and these sorts of features could be added. For the trial structure ideally you would give starts and stop times in the general frame of reference (the whole concatenated session). You could add a start time for instance for when the next recording starts. You can get the start times of each session if each session is in a different folder - `ops['frames_per_folder']` will then tell you how many frames per session.

This is relatively straight-forward to support in the main GUI, and a pull request has been started for this: https://github.com/MouseLand/suite2p/pull/223 

what do you mean with feature 3?

**Comment by zcq0730** [comment]
@carsen-stringer 
That is amazing! Thank you very much for your prompt response. Your description of the ways of getting starts and stop times information is exactly the same as how we collected the metadata information. 

And, in addition, for each trial within a session, we have additional information on the trial types, such as Hit trial, Miss trial, False alarm trial, or Correct rejection trial. Ideally, if dropbox for a filter on the trial types could be implemented would be even better. (Not really important to have this, if it is too complicated to implement. Also this might also not apply to other people's experiment designs.)

For Feature 3, sorry, the expression wasn't very clear. I made a screenshot of the GUIs, what I meant is that within the GUIs, the X axis range for label 2, by default is the whole length of concatenated tiff files. I wish it will be automatically updated to the same as X axis range of label 1. Once I selected a special trial, or when I manually moved the position of label 3, the X axis of Label 1 and label 2 could be automatically updated immediately. 

Thanks a lot! :)

![Range_Sync1](https://user-images.githubusercontent.com/40316783/82766250-cc79dc80-9e1d-11ea-84db-bc1da7e8f06c.png)
![Range_Sync2](https://user-images.githubusercontent.com/40316783/82766252-cd127300-9e1d-11ea-87db-d42b62446612.png)
